### PR TITLE
use ApplicationContextRunner to init test property values

### PR DESCRIPTION
**Pull request by sophiaso**

### CODE REVIEW

File: azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/servicebus/ServiceBusAutoConfigurationTest.java [hunk 1]
```diff
@@ -9,158 +9,84 @@
...
-                "Connection String cannot be parsed.", QueueClient.class);
+        this.contextRunner.withPropertyValues(Constants.CONNECTION_STRING_PROPERTY, Constants.INVALID_CONNECTION_STRING)
+                .withPropertyValues(Constants.QUEUE_NAME_PROPERTY, Constants.QUEUE_NAME)
+                .withPropertyValues(Constants.QUEUE_RECEIVE_MODE_PROPERTY, Constants.QUEUE_RECEIVE_MODE.name());
 
-        System.clearProperty(Constants.CONNECTION_STRING_PROPERTY);
-        System.clearProperty(Constants.QUEUE_NAME_PROPERTY);
-        System.clearProperty(Constants.QUEUE_RECEIVE_MODE_PROPERTY);
+        this.contextRunner.run((context) -> {
+            assertThat(context).doesNotHaveBean(QueueClient.class);
```
Comment (Incarnation-p-lee) [inline azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/servicebus/ServiceBusAutoConfigurationTest.java]: Can we put the only one statement in one line and remove `{}` of lambda ?

File: azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/servicebus/ServiceBusAutoConfigurationTest.java [hunk 2]
```diff
@@ -9,158 +9,84 @@
...
 
-        verifyBeanCreationException("Failed to instantiate [com.microsoft.azure.servicebus.TopicClient]: " +
-                "Factory method 'topicClient' threw exception; nested exception is " +
-                "com.microsoft.azure.servicebus.primitives.IllegalConnectionStringFormatException: " +
-                "Connection String cannot be parsed.", TopicClient.class);
-
-        System.clearProperty(Constants.CONNECTION_STRING_PROPERTY);
-        System.clearProperty(Constants.TOPIC_NAME_PROPERTY);
+        this.contextRunner.run((context) -> {
+            assertThat(context).doesNotHaveBean(TopicClient.class);
```
Comment (Incarnation-p-lee) [inline azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/servicebus/ServiceBusAutoConfigurationTest.java]: the same as above.

File: azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/servicebus/ServiceBusAutoConfigurationTest.java (hunk 1 above)
Comment (sophiaso) [reply azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/servicebus/ServiceBusAutoConfigurationTest.java]: Braces removed.

File: azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/servicebus/ServiceBusAutoConfigurationTest.java (hunk 2 above)
Comment (sophiaso) [reply azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/servicebus/ServiceBusAutoConfigurationTest.java]: Updated.

File: azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/servicebus/ServiceBusAutoConfigurationTest.java [hunk 3]
```diff
@@ -9,158 +9,76 @@
...
+                .withPropertyValues(Constants.TOPIC_NAME_PROPERTY, Constants.TOPIC_NAME)
+                .withPropertyValues(Constants.SUBSCRIPTION_NAME_PROPERTY, Constants.SUBSCRIPTION_NAME)
+                .withPropertyValues(Constants.SUBSCRIPTION_RECEIVE_MODE_PROPERTY,
+                        Constants.SUBSCRIPTION_RECEIVE_MODE.name());
 
-            assertThat(exception).isNotNull();
-            assertThat(exception.getMessage()).contains(message);
-            assertThat(exception).isExactlyInstanceOf(BeanCreationException.class);
-        }
+        this.contextRunner.run((context) -> assertThat(context).doesNotHaveBean(SubscriptionClient.class));
```
Comment (warrenzhu25) [inline azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/servicebus/ServiceBusAutoConfigurationTest.java]: `(context)` could be `context`

File: azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/servicebus/ServiceBusAutoConfigurationTest.java [hunk 4]
```diff
@@ -9,158 +9,76 @@
...
-
-        System.clearProperty(Constants.CONNECTION_STRING_PROPERTY);
-        System.clearProperty(Constants.TOPIC_NAME_PROPERTY);
-        System.clearProperty(Constants.SUBSCRIPTION_NAME_PROPERTY);
-        System.clearProperty(Constants.SUBSCRIPTION_RECEIVE_MODE_PROPERTY);
+        this.contextRunner.run((context) -> assertThat(context).doesNotHaveBean(SubscriptionClient.class));
     }
 
     @Test
     public void cannotAutowireSubscriptionClientWithInvalidCredential() {
```
Comment (warrenzhu25) [inline azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/servicebus/ServiceBusAutoConfigurationTest.java]: I suppose invalid credential should throw exception. It should be asserted also.

File: azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/servicebus/ServiceBusAutoConfigurationTest.java (hunk 3 above)
Comment (sophiaso) [reply azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/servicebus/ServiceBusAutoConfigurationTest.java]: Removed parentheses.

**Pull request merged**